### DISCUSSION TITLE
Is it necessary to install Keploy CLI?

### DISCUSSION THREAD

**Discussion by its-kunal** [keploy/keploy#1786] [Q&A] [answered]
Hey Folks,
I enjoy contributing to this amazing open-source project. On my way to learn more about this project, installing the Keploy CLI tool using WSL for Windows machines might not be necessary.

1. We can use a wizard according to user preferences to create the docker-compose file and the docker-compose command. as done by Weaviate - [Link](https://weaviate.io/developers/weaviate/installation/docker-compose#configurator).
2. Installing CLI may create an entry barrier for most folks.

**Comment by SkySingh04** [comment]
Hey there @its-kunal! We are currently planning on developing extensions for various popular IDE's such as VsCode and IDEA for Keploy, which I think would be a better solution than a Wizard. At the end, Keploy is a CLI tool, hence it wouldn't make sense for an installation wizard, considering Keploy has One-Click Installation.

What are your opinions on this?

**Reply by its-kunal** [reply]
Yes, I am well-versed in the fact that there is development of Keploy extension and also appreciate it. Using a wizard will help to containerize the application along with Keploy without using methods like Docker-in-Docker (DnD) and run Keploy in any machine in just one command.

**Comment by Sarthak160** [comment, marked as answer]
Hi @its-kunal , Currently Keploy is following cli based architecture. If you can share some more details that how would you approach on the above. It will be helpful to know, if its really relevant or overkill for now.
